ui/selectors/selectors.js

```diff
@@ -151,7 +154,26 @@ export function getNetworkIdentifier(state) {
 
 export function getCurrentChainId(state) {
   const { chainId } = getProviderConfig(state);
-  return chainId;
+
+  if (state.metamask.domains === undefined) {
+    console.info(
+      `[getCurrentChainId] state.metamask.domains is undefined, returning global chainId (${chainId})`,
+    );
+    return chainId;
+  }
+
+  const pageChainId =
+    state.metamask.domains[state.activeTab?.origin || ORIGIN_METAMASK];
```

Comment:
This returns a `networkClientId`, not `chainId`, so we'll need to do a lookup based on the `networkClientId`

Reply:
Fixed in second commit.